internal/providers/github/github.go

```diff
@@ -115,3 +126,78 @@ func ParseV1Config(rawCfg json.RawMessage) (*minderv1.GitHubProviderConfig, erro
 
 	return w.GitHub, nil
 }
+
+func (c *RestClient) waitIfRateLimited(ctx context.Context) error {
+	blockedChan, ok := c.tokenStatStore.GetBlockingChan(c.token, db.ProviderTypeGithub)
+	if !ok {
+		return nil
+	}
+
+	select {
+	case <-blockedChan:
+		return nil
+	case <-ctx.Done():
+		return ctx.Err()
+	case <-time.After(MaxRateLimitWait):
+		return fmt.Errorf("rate limit wait exceeded for %s", c.GetOwner())
+	}
+}
+
+func (c *RestClient) checkRateLimitResponse(ctx context.Context, resp *http.Response) error {
+	if resp.StatusCode != http.StatusForbidden && resp.StatusCode != http.StatusTooManyRequests {
```

Comment:
Are there cases where `resp` might be `nil` on a fatal error in which case we'd have to just rely on the error rather then the reply? (I noticed that there's no `err` check between calling `c.client.Issues.CreateComment` and dereferencing `resp`)

```
	_, resp, err := c.client.Issues.CreateComment(ctx, owner, repo, number, &github.IssueComment{
			Body: &comment,
		})

		rateLimCheckErr := c.checkRateLimitResponse(ctx, resp.Response)
		if rateLimCheckErr != nil {
			return rateLimCheckErr
		}
```